design-system/react/src/components/Button/Button.tsx

```diff
@@ -22,7 +24,7 @@ export function createIcon(
   return typeof icon === "string" ? (
     <Icon icon={icon} label={iconAriaLabel} size={iconSize} />
   ) : (
-    icon && createElement(icon, { label: iconAriaLabel })
+    icon && cloneElement(icon as any, { label: iconAriaLabel })
```

Comment:
it has to be as any?

Reply:
I think because `icon` has two so much different types that it can be (`string | ReactNode`) the `cloneElement` complain about receiving the prop here, I tried to solve this but was getting too much time... I can let as a technical debit